functions/src/api/middlewares/authenticateJWTMiddleware.spec.ts

```diff
@@ -0,0 +1,127 @@
+// tslint:disable-next-line
+import '../../testUtils/mockFirebaseAdminAndFunctions'
+import { firestoreGet } from '../../testUtils/firestoreStub'
+
+import { authenticateJWTMiddleware } from './authenticateJWTMiddleware'
+import {
+    ForbiddenError,
+    PreconditionFailedError,
+    UnauthorizedError,
+} from '../errors/HttpError'
+
+describe('authentificateJWTMiddleware', () => {
+    const mockNext = jest.fn()
+    const mockRes = jest.fn()
+
+    afterEach(() => {
+        mockNext.mockClear()
```

Comment:
if mocks are cleared in `afterEach`, shouldn't they be created in a `beforeEach` callback ?

Reply:
they are created once for all the test file, just the count on the mock are reset after each test to prevent polution.